src/main/java/com/example/housingmanagement/api/services/HouseService.java

```diff
@@ -45,42 +45,49 @@ public boolean existsByHouse(HouseRequest houseRequest) {
     }
 
     public boolean houseHasSpareCapacity(HouseRequest houseRequest) {
-        Optional<HouseInternalEntity> houseInternalEntityToBeChecked = Optional.ofNullable(houseRepository.findByHouseNumber(houseRequest.getHouseNumber()));
+        Optional<HouseInternalEntity> houseInternalEntityToBeChecked =
+                Optional.ofNullable(houseRepository.findByHouseNumber(houseRequest.getHouseNumber()));
 
-        return houseInternalEntityToBeChecked.get().getCurrentCapacity() < houseInternalEntityToBeChecked.get().getMaxCapacity();
+        return houseInternalEntityToBeChecked
+                .filter(houseInternalEntity -> houseInternalEntity.getCurrentCapacity() < houseInternalEntity.getMaxCapacity()).isPresent();
     }
 
     public void increaseHouseCurrentCapacityByOne(HouseRequest houseRequest) {
 
         Optional<HouseInternalEntity> houseInternalEntityToIncreaseCapacityByOne =
                 Optional.ofNullable(houseRepository.findByHouseNumber(houseRequest.getHouseNumber()));
 
-        houseInternalEntityToIncreaseCapacityByOne.get().setCurrentCapacity(houseInternalEntityToIncreaseCapacityByOne.get().getCurrentCapacity() + 1);
+        houseInternalEntityToIncreaseCapacityByOne
+                .ifPresent(houseInternalEntity -> houseInternalEntity.setCurrentCapacity(houseInternalEntity.getCurrentCapacity() + 1));
 
-        HouseInternalEntity houseInternalEntity = new HouseInternalEntity(
-                houseInternalEntityToIncreaseCapacityByOne.get().getId(),
-                houseInternalEntityToIncreaseCapacityByOne.get().getHouseNumber(),
-                houseInternalEntityToIncreaseCapacityByOne.get().getMaxCapacity(),
-                houseInternalEntityToIncreaseCapacityByOne.get().getCurrentCapacity());
+        if (houseInternalEntityToIncreaseCapacityByOne.isPresent()) {
+            HouseInternalEntity houseInternalEntity = new HouseInternalEntity(
+                    houseInternalEntityToIncreaseCapacityByOne.get().getId(),
```

Comment:
houseInternalEntityToIncreaseCapacityByOne.get() x milion

Reply:
to samo, czyli introduce variable